test/int/builds-external-ddoc.js

```diff
@@ -0,0 +1,166 @@
+    author: 'int. test',
+    node_modules: []
+  };
+
+  return doc;
+};
+
+describe('"Builds External" Design document', () => {
+  before(() => resetDb());
+  afterEach(() => clearDb());
+
+  describe('validate_doc_update for anonymous users', () => {
+    it('we should be using an anonymous user', async () => {
+      const serverUrl = new URL(buildsDb.name);
+      serverUrl.pathname = '';
+      const session = await (await PouchDB.fetch(`${serverUrl}/_session`)).json();
+      expect(session.userCtx).to.deep.equal({ name: null, roles: [] });
+    });
+
+    it('should allow to push builds', async () => {
+      await buildsDb.put(withBuildInfo({_id: 'medic:medic:a-branch', value: 1}));
+    });
+
+    it('should allow pushing the same build multiple times', async () => {
+      const buildId = 'medic:medic:a-branch';
+      const res = await buildsDb.put(withBuildInfo({ _id: buildId, value: 1 }));
+      await buildsDb.put(withBuildInfo({ _id: buildId, value: 2, _rev: res.rev }));
+      const doc = await buildsDb.get(buildId);
+      expect(doc.value).to.equal(2);
+    });
+
+    it('should allow pushing the same "release" multiple times', async () => {
+      const buildId = 'medic:medic:2.0.0';
+      const { rev } = await buildsDb.put(withBuildInfo({ _id: buildId, value: 1 }));
+      await buildsDb.put(withBuildInfo({ _id: buildId, value: 2, _rev: rev }));
+      const doc = await buildsDb.get(buildId);
+      expect(doc.value).to.equal(2);
+    });
+
+    it('should allow pushing the same "pre-release" multiple times', async () => {
+      const buildId = 'medic:medic:2.0.0-beta.1';
+      const { rev } = await buildsDb.put(withBuildInfo({ _id: buildId, value: 1 }));
+      await buildsDb.put(withBuildInfo({ _id: buildId, value: 2, _rev: rev }));
+      const doc = await buildsDb.get(buildId);
+      expect(doc.value).to.equal(2);
+    });
+
+    it('should allow deleting branches', async () => {
+      const buildId = 'medic:medic:a-branch';
+      const { rev } = await buildsDb.put(withBuildInfo({ _id: buildId, value: 1 }));
+      await buildsDb.remove({ _id: buildId, _rev: rev });
+    });
+
+    it('should allow deleting releases', async () => {
+      const buildId = 'medic:medic:2.0.0';
+      const { rev } = await buildsDb.put(withBuildInfo({ _id: buildId, value: 1 }));
+      await buildsDb.remove({ _id: buildId, _rev: rev });
+    });
+
+    it('should allow deleting pre-releases', async () => {
+      const buildId = 'medic:medic:2.0.0-beta.1';
+      const { rev } = await buildsDb.put(withBuildInfo({ _id: buildId, value: 1 }));
+      await buildsDb.remove({ _id: buildId, _rev: rev });
+    });
+
+    it('should now allow creation of documents with invalid ids', async () => {
+      try {
+        await buildsDb.put(withBuildInfo({ _id: 'not-a-valid-version-identifier' }));
+        expect.fail('Should have thrown');
+      } catch (err) {
+        expect(err.message).to.equal('Document _id format invalid');
+      }
```

Comment:
I'd like to use chai-as-promised and do this with `to.be.rejectedWith` to remove the try/catch business here. Given that's not in our codebase yet, I'll skip on it for now and leave this alone, but I really would like to get this (and all the other cases in our tests in general) to be logical chai checks and not have the try/catch here.

Reply:
Turns out chai-as-promised was already in the package :)

Reply:
Oh, interesting. I grepped for it, but must have done something wrong. Anyway, do my chai-as-promised sales pitch in the future :)